iOS4 - HW21 - Виноградов Дмитрий.playground/Contents.swift

```diff
@@ -0,0 +1,140 @@
+import Foundation
+
+// MARK: - UNTOUCHABLE
+
+public struct Chip {
+    public enum ChipType: UInt32 {
+        case small = 1
+        case medium
+        case big
+    }
+    
+    public let chipType: ChipType
+    
+    public static func make() -> Chip {
+        guard let chipType = Chip.ChipType(rawValue: UInt32(arc4random_uniform(3) + 1)) else {
+            fatalError("Incorrect random value")
+        }
+        
+        return Chip(chipType: chipType)
+    }
+    
+    public func sodering() {
+        let soderingTime = chipType.rawValue
+        sleep(UInt32(soderingTime))
+    }
+}
+
+// MARK: - Thread-Safety array
+
+public class WorkArray<T> {
+    private var array: [T] = []
+    private let accessQueue = DispatchQueue(label: "SynchronizedArrayAccess", attributes: .concurrent)
+    
+    public func append(newElement: T) {
+        
+        self.accessQueue.async(flags:.barrier) {
+            self.array.append(newElement)
+        }
```

Comment:
В асинхронном коде self нужно защищать
```suggestion
       accessQueue.async(flags:.barrier) { [unowned self] in
            self.array.append(newElement)
        }
```